src/main/java/net/clgd/ccemux/plugins/builtin/CCEmuXAPI.java

```diff
@@ -50,7 +54,33 @@ public API(CCEmuX emu, EmulatedComputer computer, String name) {
 
 			methods.put("openEmu", o -> {
 				int id = ArgumentHelper.optInt(o, 0, -1);
-				EmulatedComputer ec = emu.createComputer(b -> b.id(id));
+				String program = ArgumentHelper.optString(o, 1, null);
+				
+				EmulatedComputer ec = emu.createComputer(b -> b.id(id), program == null);
+				
+				if (program != null) {
+					IWritableMount mount = ec.getRootMount();
+					
+					try {
+						if (!mount.exists(program)) {
+							emu.removeComputer(ec);
+							throw new LuaException("program not found");
+						}
+						
+						if (!mount.isDirectory("startup/")) mount.makeDirectory("startup");
```

Comment:
If there's a legacy `startup` file it will be deleted with no warning here.

Reply:
@apemanzilla how does it get deleted? i dont see anything that deletes a legacy startup file

Reply:
You can't have two entries with the same name. If you have a `startup` file, it will be replaced by a `startup` directory.